packages/react-native/scripts/cocoapods/privacy_manifest_utils.rb

```diff
@@ -0,0 +1,159 @@
+        installer.pod_targets.each do |pod_target|
+            # puts pod_target
+            pod_target.file_accessors.each do |file_accessor|
+            file_accessor.resource_bundles.each do |bundle_name, bundle_files|
+                bundle_files.each do |file_path|
+                # This needs to be named like that due to apple requirements
+                if File.basename(file_path) == 'PrivacyInfo.xcprivacy'
+                    content = Xcodeproj::Plist.read_from_path(file_path)
+                    accessed_api_types = content["NSPrivacyAccessedAPITypes"]
+                    accessed_api_types.each do |accessed_api|
+                    api_type = accessed_api["NSPrivacyAccessedAPIType"]
+                    reasons = accessed_api["NSPrivacyAccessedAPITypeReasons"]
+                    used_apis[api_type] ||= []
+                    used_apis[api_type] += reasons
+                    end
+                end
+                end
+            end
+            end
+        end
+        return used_apis
+    end
+
+    def self.get_privacy_manifest_paths_from(user_project)
+        privacy_manifests = user_project
+            .files
+            .select { |p|
+                p.path&.end_with?('PrivacyInfo.xcprivacy')
+            }
+        return privacy_manifests
+    end
+
+    def self.get_core_accessed_apis()
+        file_timestamp_accessed_api = {
+            "NSPrivacyAccessedAPIType" => "NSPrivacyAccessedAPICategoryFileTimestamp",
+            "NSPrivacyAccessedAPITypeReasons" => ["C617.1"],
+        }
+        user_defaults_accessed_api = {
+            "NSPrivacyAccessedAPIType" => "NSPrivacyAccessedAPICategoryUserDefaults",
+            "NSPrivacyAccessedAPITypeReasons" => ["CA92.1"],
+        }
+        boot_time_accessed_api = {
+            "NSPrivacyAccessedAPIType" => "NSPrivacyAccessedAPICategorySystemBootTime",
+            "NSPrivacyAccessedAPITypeReasons" => ["35F9.1"],
+        }
+        return [file_timestamp_accessed_api, user_defaults_accessed_api, boot_time_accessed_api]
+    end
+
+
+    def self.get_user_project_from(installer)
+        user_project = installer.aggregate_targets
+                    .map{ |t| t.user_project }
+                    .first
+        return user_project
+    end
+
+    def self.add_privacy_manifest_if_needed(installer)
+        user_project = get_user_project_from(installer)
+        privacy_manifest = self.get_privacy_manifest_paths_from(user_project).first
+        if privacy_manifest.nil?
+            privacy_manifest = {
+                "NSPrivacyCollectedDataTypes" => [],
+                "NSPrivacyTracking" => false,
+                "NSPrivacyAccessedAPITypes" => get_core_accessed_apis
+            }
+            path = File.join(user_project.path.parent, "PrivacyInfo.xcprivacy")
+            Xcodeproj::Plist.write_to_path(privacy_manifest, path)
```

Comment:
I guess it depends on what we want this logic to do:

1) Place the manifest outside of the project, and inform the user that they need to copy it manually and add their reasons? – we don't want to reference it if we don't place it in the project.

2) Create a manifest with RN reasons inside the project (so not user_project.path.**parent**) and auto reference it?
If so, perhaps we can use the same `add_aggregated_privacy_manifest` method and just not do aggregation to simplify the code somewhat.

Since this code path runs only if a user opts-out of the aggregation step, I'd keep it as simple as possible, do 1), and treat it as a "_hey, we noticed you don't have a manifest and decided to opt-out of aggregation, here's a sample file you could use that we placed in your project root but didn't touch your actual ios project_".